src/diffusers/models/unets/unet_i2vgen_xl.py

```diff
@@ -543,7 +518,8 @@ def forward(
             forward_upsample_size = True
 
         # 1. time
-        timesteps = _to_tensor(timestep, sample.device)
+        timesteps = timestep
+        timesteps = timesteps.to(sample.device)
```

Comment:
Instead of using `_to_tensor()` here we simply do the device placement since `timestep` here will always come as a tensor IIUC.

Reply:
Fixed in 3d26814f. 

If a scheduler in the pipeline changes and it `set_timesteps()` doesn't ensure tensor packing, it's gonna fail. I think the bigger question might be to think whether we should start setting `timesteps` to always be tensors from `set_timesteps()`. But I also think it's not a super high-prio right now.